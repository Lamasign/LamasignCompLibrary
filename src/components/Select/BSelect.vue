<template>
  <div class="space-y-1 w-full h-10">
    <label
      id="listbox-label"
      class="block text-sm leading-5 font-medium text-gray-700 h-full"
    >
      <slot name="SelectLabel"></slot>
    </label>

    <div
      class="relative h-full"
      v-click-outside="closeDropdown"
    >
      <span class="inline-block w-full rounded-md shadow-sm h-full">
        <button
          type="button"
          @click="openDropdown"
          aria-haspopup="listbox"
          aria-expanded="true"
          aria-labelledby="listbox-label"
          class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue-300 transition ease-in-out duration-150 sm:text-sm sm:leading-5"
        >
          <div class="flex items-center space-x-3 h-full">
            <span class="block truncate">
              {{ updatedValue ? updatedValue[field] : 'SÃ©lectionnez'}}
            </span>
          </div>
          <span class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none">
            <svg
              class="h-5 w-5 text-gray-400"
              viewBox="0 0 20 20"
              fill="none"
              stroke="currentColor"
            >
              <path
                d="M7 7l3-3 3 3m0 6l-3 3-3-3"
                stroke-width="1.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
      </span>

      <!-- Select popover, show/hide based on select state. -->
      <div
        v-show="isOpen"
        class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      >
        <ul
          tabindex="-1"
          role="listbox"
          aria-labelledby="listbox-label"
          aria-activedescendant="listbox-item-3"
          class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        >

          <!--
            Select option, manage highlight styles based on mouseenter/mouseleave and keyboard navigation.
            Highlighted: "text-white bg-indigo-600", Not Highlighted: "text-gray-900"
          -->
          <li
            tabindex="0"
            @click="select(item)"
            id="listbox-item-0"
            role="option"
            v-for="(item, index) in data"
            :key="index"
            class="text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white hover:bg-indigo-600 focus:outline-none focus:text-white focus:bg-indigo-600"
          >
            <div class="flex items-center space-x-3">
              <!-- Selected: "font-semibold", Not Selected: "font-normal" -->
              <span
                class="block truncate"
                :class="{ 'font-normal' : !isSelected(item).value , 'font-semibold' : isSelected(item).value}"
              >
                {{ item[field] }}
              </span>
            </div>

            <!--
              Checkmark, only display for selected option.
              Highlighted: "text-white", Not Highlighted: "text-indigo-600"
            -->
            <span
              v-show="isSelected(item).value"
              class="absolute inset-y-0 right-0 flex items-center pr-4"
            >
              <svg
                class="h-5 w-5"
                viewBox="0 0 20 20"
                fill="currentColor"
              >
                <path
                  fill-rule="evenodd"
                  d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                  clip-rule="evenodd"
                />
              </svg>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</template>
<script lang='ts'>
import { computed, defineComponent, ref } from '@vue/composition-api'

export default defineComponent({
  name: 'BSelect',
  props: {
    data: {
      type: Array,
      required: true,
    },
    value: {
      type: [String, Object, Number],
      required: true,
    },
    open: {
      type: Boolean,
      default: false,
    },
    field: {
      type: String,
      required: true,
    },
  },
  setup(props, { emit }) {
    const isOpen = ref(props.open)
    const updatedValue = ref<unknown>(props.value)

    const isSelected = (el: unknown) => computed(() => updatedValue.value === el)

    function closeDropdown() {
      isOpen.value = false
      emit('open', false)
    }

    function openDropdown() {
      isOpen.value = true
      emit('open', true)

    }

    function select(newValue: unknown) {
      updatedValue.value = newValue
      emit('valueSelect', newValue)
      closeDropdown()
    }

    return {
      isOpen,
      closeDropdown,
      openDropdown,
      isSelected,
      select,
      updatedValue,
    }
  },
})
</script>