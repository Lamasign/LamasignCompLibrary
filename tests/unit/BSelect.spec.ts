import { SelectColorsArray, SelectColorsEnum } from './../../src/components/Select/BSelect.types'
import BSelect from '@/components/Select/BSelect.vue'
import { createWrapper } from './utils'

describe('BSelect.vue', () => {
	const list = [
		{
			value: false,
			name: 'Frozen Yogurt',
			calories: 159,
			fat: 6.0,
			carbs: 24,
			protein: 4.0,
			iron: '1%'
		},
		{
			value: false,
			name: 'Ice cream sandwich',
			calories: 237,
			fat: 9.0,
			carbs: 37,
			protein: 4.3,
			iron: '1%'
		},
		{
			value: false,
			name: 'Eclair',
			calories: 262,
			fat: 16.0,
			carbs: 23,
			protein: 6.0,
			iron: '7%'
		},
		{
			value: true,
			name: 'Cupcake',
			calories: 305,
			fat: 3.7,
			carbs: 67,
			protein: 4.3,
			iron: '8%'
		},
		{
			value: false,
			name: 'Gingerbread',
			calories: 356,
			fat: 16.0,
			carbs: 49,
			protein: 3.9,
			iron: '16%'
		},
		{
			value: false,
			name: 'Jelly bean',
			calories: 375,
			fat: 0.0,
			carbs: 94,
			protein: 0.0,
			iron: '0%'
		},
		{
			value: false,
			name: 'Lollipop',
			calories: 392,
			fat: 0.2,
			carbs: 98,
			protein: 0,
			iron: '2%'
		},
		{
			value: false,
			name: 'Honeycomb',
			calories: 408,
			fat: 3.2,
			carbs: 87,
			protein: 6.5,
			iron: '45%'
		},
		{
			value: false,
			name: 'Donut',
			calories: 452,
			fat: 25.0,
			carbs: 51,
			protein: 4.9,
			iron: '22%'
		},
		{
			value: false,
			name: 'KitKat',
			calories: 518,
			fat: 26.0,
			carbs: 65,
			protein: 7,
			iron: '6%'
		}
	]

	it('render the content of his default slot', () => {
		expect.assertions(1)
		const slotContent = 'Select'
		const wrapper = createWrapper(BSelect, {
			slots: { SelectLabel: slotContent },
			propsData: {
				value: list[0],
				field: 'name',
				data: list,
			},
		})
		expect(wrapper.text()).toMatch(slotContent)
	})

	it('renders with the appropriate classes based on the "outlined" prop', async () => {
		expect.assertions(SelectColorsArray.length + 6)

		const defaultWrapper = createWrapper(BSelect, {
			propsData: {
				value: list[0],
				field: 'name',
				data: list,
			}
		})
		expect(defaultWrapper.element).toMatchSnapshot()

		await Promise.all(SelectColorsArray.map(async (color) => {
			const wrapperWithProps = createWrapper(BSelect, {
				propsData: {
					color,
					value: list[0],
					field: 'name',
					data: list,

				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
			return Promise.resolve(expect(wrapperWithProps.element).toMatchSnapshot())
		}))
	})

	it('emits a click "event" when the Select is clicked', async () => {
		const wrapper = createWrapper(BSelect, {
			propsData: {
				data: list,
				field: 'name',
				value: list[0],
			}
		})
		await wrapper.find('#openButton').trigger('click')
		expect(wrapper.emitted().open).toBeTruthy()
		expect(wrapper.find('div ul #listbox-label'))

		await wrapper.find('div ul li ').trigger('click')
		expect(wrapper.emitted().close).toBeTruthy()
	})
})