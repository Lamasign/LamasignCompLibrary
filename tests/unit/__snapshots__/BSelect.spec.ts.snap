// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 1`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-green focus:bg-green"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            SÃ©lectionnez
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 2`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-blue focus:bg-blue"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 3`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-blue focus:bg-blue"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 4`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-orange focus:bg-orange"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 5`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-orange focus:bg-orange"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 6`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-red focus:bg-red"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 7`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-red focus:bg-red"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 8`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-green focus:bg-green"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 9`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-green focus:bg-green"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 10`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-yellow focus:bg-yellow-500"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`BSelect.vue renders with the appropriate classes based on the "outlined" prop 11`] = `
<div
  class="space-y-1 w-full h-10"
>
  <label
    class="block text-sm leading-5 font-medium text-gray-700 h-full"
    id="listbox-label"
  />
   
  <div
    class="relative h-full"
  >
    <span
      class="inline-block w-full rounded-md shadow-sm h-full"
    >
      <button
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-labelledby="listbox-label"
        class="cursor-pointer relative w-full h-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue transition ease-in-out duration-150 sm:text-sm sm:leading-5 text-gray-900 select-none relative py-2 pl-3 pr-9  cursor-pointer hover:text-white focus:outline-none focus:text-white hover:bg-yellow focus:bg-yellow-500"
        id="openButton"
        type="button"
      >
        <div
          class="flex items-center space-x-3 h-full"
        >
          <span
            class="block truncate"
          >
            
            
          
          </span>
        </div>
         
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              d="M7 7l3-3 3 3m0 6l-3 3-3-3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </span>
      </button>
    </span>
     
    <div
      class="absolute mt-1 w-full rounded-md bg-white shadow-lg"
      style="display: none;"
    >
      <ul
        aria-activedescendant="listbox-item-3"
        aria-labelledby="listbox-label"
        class="max-h-56 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
